### PR TITLE
Officially introduce amalgamated builds

### PR DESCRIPTION
**Pull request by akosthekiss**
Remove redundancy between all-in-one and all-in-one-source builds
by keeping only the second, and adopt the more established term
"amalgamated" build for it. This change includes the following:

- Replace `ENABLE_ALL_IN_ONE` and `ENABLE_ALL_IN_ONE_SOURCE` cmake
  options with `ENABLE_AMALGAM` top-level option.
- Replace `--all-in-one` option of `build.py` helper with
  `--amalgam`.
- Merge the `srcmerger.py` and `srcgenerator.py` tool scripts into
  `amalgam.py` (with improvements).
- Update documentation.

JerryScript-DCO-1.0-Signed-off-by: Akos Kiss akiss@inf.u-szeged.hu

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -109,9 +110,11 @@ endif()
 
 if(USING_MSVC)
   set(JERRY_MATH         OFF)
+  set(ENABLE_AMALGAM     ON) # FIXME: This should not be needed but right now it is. To be tracked down and followed up.
```
Comment (lygstate) [inline CMakeLists.txt:113]: Fixed by https://github.com/jerryscript-project/jerryscript/pull/4392

**Review by rerobika — APPROVED** [review]
LGTM

**Comment by lygstate** [comment]
LGTM

**Comment by rerobika** [comment]
Please rebase.

**Comment by akosthekiss** [comment]
Rebased.

**Comment by LaszloLango** [comment]
Is this related to #4406?

**Comment by akosthekiss** [comment]
@LaszloLango Partially. That issue brought the two versions of all-in-one builds into focus. Analysis later revealed that two different approaches were maintained in parallel. Plus, one of them was completely undocumented. Finally came the name change (potentially more widely known and/or accepted -- even though that may be arguable).

**Review by dbatyai — APPROVED** [review]
LGTM

**Pull request merged**